python/tank/platform/engine.py

```diff
@@ -1832,14 +1836,69 @@ def start_engine(engine_name, tk, context):
     Raises TankEngineInitError if an engine could not be started
     for the passed context.
     """
+    return _start_engine(engine_name, tk, None, context)
+
+
+def _restart_engine(new_context):
+
+    engine = current_engine()
+    try:
+        # Track some of the current state before restarting the engine.
+        old_context = engine.context
+
+        # Restart the engine. If we were given a new context to use,
+        # use it, otherwise restart using the same context as before.
+        new_context = new_context or engine.context
```

Comment:
Yeah, I thought about that. I felt it was an abuse of language as well. I kept in mind that the reason people asked for this was because they were depending on the engine reinit to detect context change. Maybe context change is not the correct term for this new hook. Maybe it's context reset or something. It's not engine init, because the engine is not always torn down and it's not context change per se, since as you've pointed out, the context can stay the same.